yetanotherco/aligned_layer
# fix: accumulated aggregator cost panels

# Fix Accumulated Aggregator Cost Panels

## Description

This PR addresses a couple of issues in the Aggregator cost panels.
- Separates the panels into historical and time ranges.
- Uses the receipt values instead of the raw transaction ones (we were overestimating the costs since not all of the gas limit is consumed during a transaction execution).
- Adds a target to clean the metrics db.

![image](https://github.com/user-attachments/assets/0860b5bb-748a-4eee-ba59-06d7a3376a5b)

## Type of change


## How to test
1. Start all Aligned components and start sending proofs
2. Start Grafana and go to [localhost:3000](localhost:3000):
```bash
make run_metrics
```
3. Check the new panels and play with the time range selector on the top-right corner:
![image](https://github.com/user-attachments/assets/e850a6a5-0172-4a30-bc16-4d34b50f5983)
4. Use the new target to clean the db:

```bash
make metrics_clean_db
```

## Checklist

  - [Link to the open issue addressing it]()
